src/Services/Device.vala

```diff
@@ -223,6 +223,14 @@ public class Power.Services.Device : Object {
         properties_updated ();
     }
 
+    private Power.Services.Device.Type determine_device_type () {
+        // In case a all-in-one keyboard is clasified as mouse because of a mouse pointer. we should show it as keyboard.
```

Comment:
can we include a link to the upstream issue here, so that the next person who encounters this knows to check if the issue has been resolved upstream: https://gitlab.freedesktop.org/upower/upower/-/issues/139